lib/concurrent/executor/ruby_thread_pool_executor.rb

```diff
@@ -123,142 +122,231 @@ def queue_length
     #
     # @return [Integer] the remaining_capacity
     def remaining_capacity
-      mutex.synchronize { @max_queue == 0 ? -1 : @max_queue - @queue.length }
+      mutex.synchronize do
+        if ns_limited_queue?
+          @max_queue - @queue.length
+        else
+          -1
+        end
+      end
     end
 
-    # Returns an array with the status of each thread in the pool
-    #
-    # This method is deprecated and will be removed soon.
-    def status
-      warn '[DEPRECATED] `status` is deprecated and will be removed soon.'
-      mutex.synchronize { @pool.collect { |worker| worker.status } }
+    alias_method :current_length, :length # TODO remove?
```

Comment:
I don't remember why I created this alias in the first place, but I've come to the realization that I used to make far too many aliases. I'm :ok: to remove it.